### PR TITLE
Support saving the metric report

### PR DESCRIPTION
**Pull request by lolipopshock**
```
pawls metric <labeling_folder> <config_file> \
    --textual-categories "xx" --non-textual-categories "xx" \
    --save <save-folder>
```

### CODE REVIEW

File: cli/pawls/commands/metric.py [hunk 1]
```diff
@@ -343,10 +367,13 @@ def metric(
             coco_results, coco_category_results = coco_eval.calculate_ap_scores()
 
             if verbose:
-                coco_eval.show_results(coco_results)
+                save_table = coco_eval.show_results(coco_results)
```
Comment (DeNeutoy) [inline cli/pawls/commands/metric.py:370]: This method name could be changed if it no longer shows the results

**Review by DeNeutoy — COMMENTED** [review]
So, this is a bit fiddly - really it would be good to rely on piping in the terminal to save metric reports. I assume that the csv format is slightly different to the one printed? If this is the case, this could be called `--format csv` with the idea that it makes it easier to add other output formats later on, like json. This is also more flexible, as piping would allow you to e.g append to the file, rather than overwriting the content, which I think is what this will do. Does that make sense?

File: cli/pawls/commands/metric.py (hunk 1 above)
Comment (lolipopshock) [reply cli/pawls/commands/metric.py:370]: Yes, that's a great point!

**Comment by lolipopshock** [comment]
> So, this is a bit fiddly - really it would be good to rely on piping in the terminal to save metric reports. I assume that the csv format is slightly different to the one printed? If this is the case, this could be called `--format csv` with the idea that it makes it easier to add other output formats later on, like json. This is also more flexible, as piping would allow you to e.g append to the file, rather than overwriting the content, which I think is what this will do. Does that make sense?

Thanks! That's a great point, and I also think piping should be an elegant solution to this issue. 

But there are two difficulties for piping:
1. Currently we generate some prompt for better UX. 
2. And to display the tables neatly, we use `tabulate` to prettify the tables, which makes the table results significantly different from regular csv formats. 

And the displayed IAA tables are based on dataframes, which are somewhat equivalent csv tables. 

The reason I am building the output function is that we might have a lot of annotators and the table will be huge. It would be difficult to view the table in command line and exporting and view it separately might be a better option. 

Do you have any good ideas to solve these problem? Thank you!

**Comment by DeNeutoy** [comment]
Ah ok, the first is a good point. I could see something like conditional print statements based on whether a user passes `--format`, but maybe that's overkill for now. LGTM!

**Review by DeNeutoy — APPROVED** [review]
LGTM

File: cli/pawls/commands/metric.py (hunk 1 above)
Comment (lolipopshock) [reply cli/pawls/commands/metric.py:370]: I double checked - This function still prints the table, and also returns the table. I feel like something like `print_and_savet_results` should be better - but just a bit awkward? We could come back and fix this issue later after we find a better strategy for saving the reports.

**Comment by lolipopshock** [comment]
> Ah ok, the first is a good point. I could see something like conditional print statements based on whether a user passes `--format`, but maybe that's overkill for now. LGTM!

Thanks! I think conditional printing is great - I might need to further think the API design and will come back to this issue later.

**Pull request merged**